app/controllers/search_controller.rb

```diff
@@ -10,19 +10,15 @@ def explore
 
   def search
   	sort_arr = sort_order
-  	repositories_by_attributes = Repository.where("lower(title) LIKE ?
+  	@repositories = Repository.where("lower(title) LIKE ?
                                                 OR lower(description) LIKE ?
                                                 OR lower(user_username) LIKE ?
                                                 OR lower(category) LIKE ?",
                                   "%#{params[:q].downcase}%",
                                   "%#{params[:q].downcase}%",
                                   "%#{params[:q].downcase}%",
                                   "%#{params[:q].downcase}%")
-    repositories_by_categories = []
-    Category.where('lower(name) LIKE ?', params[:q].downcase).each do |cat|
-      repositories_by_categories << cat.repository
-	  end
-    @repositories = repositories_by_attributes + repositories_by_categories
+
     @repositories = @repositories.uniq
```

Comment:
You can probably collapse all of these manipulations into 1 assignment which will prevent active record for loading _every_ repository and instead just load the 12 needed for the pagination. Something like

```
@repositories = Repository.where(stuff)distinct.paginate(bla)
```